src/ui/camera.js

```diff
@@ -803,6 +805,10 @@ class Camera extends Evented {
             options.duration = 1000 * S / V;
         }
 
+        if (options.maxDuration && options.duration > options.maxDuration) {
+            options.duration = 0;
```

Comment:
I think it would make more sense to set `options.duration = options.maxDuration` in this case.

Reply:
Thanks for the review, @mollymerp! According to the last @mourner's comment in  https://github.com/mapbox/mapbox-gl-js/issues/3904, it was decided to reset `duration` to `0`. So should I change it, or not?

Reply:
@mollymerp yeah, that was intentional — consider a use case where you need to switch to a view from another part of the world, but you don't want it to take forever to animate. Setting to `maxDuration` would make the animation too fast, since it's a very long fly path — instead, just switching to the new view in a moment is usually a better experience IMO.

Reply:
So everything is fine, then? Should I write tests and add a new example?

Reply:
@stepankuzmin definitely worth adding a test! I don't think a small option like this needs a separate example though.

Reply:
@mourner alright fair enough! I would think if a user had a case with a very long "flight path" they would rather explicitly set duration to 0 but I can definitely see both as valid 🤷‍♀️

Reply:
> I would think if a user had a case with a very long "flight path" they would rather explicitly set duration to 0
@mollymerp the user doesn't know beforehand how much the flight will take between two given positions (e.g. say you have a list of locations around the world and you click to switch between them in arbitrary order) — this is the primary reason for introducing `maxDuration` in the first place, and is similar to how other libs like Leaflet and Google Maps do it, disabling animation automatically if the flight path would be too long.